src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -767,175 +624,171 @@ static MemberName generateCustomizedCode(LambdaForm form, MethodType invokerType
     }
 
     /** Generates code to check that actual receiver and LambdaForm matches */
-    private boolean checkActualReceiver() {
+    private boolean checkActualReceiver(CodeBuilder cob) {
         // Expects MethodHandle on the stack and actual receiver MethodHandle in slot #0
-        mv.visitInsn(Opcodes.DUP);
-        mv.visitVarInsn(Opcodes.ALOAD, localsMap[0]);
-        mv.visitMethodInsn(Opcodes.INVOKESTATIC, MHI, "assertSame", LLV_SIG, false);
+        cob.stackInstruction(Opcode.DUP);
+        cob.loadInstruction(TypeKind.ReferenceType, localsMap[0]);
+        cob.invokeInstruction(Opcode.INVOKESTATIC, MHI, "assertSame", LLV_SIG, false);
         return true;
     }
 
-    static String className(String cn) {
+    static ClassDesc classDesc(String cn) {
         assert checkClassName(cn): "Class not found: " + cn;
-        return cn;
+        return ClassDesc.ofDescriptor(cn);
     }
 
     static boolean checkClassName(String cn) {
-        Type tp = Type.getType(cn);
         // additional sanity so only valid "L;" descriptors work
-        if (tp.getSort() != Type.OBJECT) {
+        if (!cn.startsWith("L") || !cn.endsWith(";")) {
             return false;
         }
         try {
-            Class<?> c = Class.forName(tp.getClassName(), false, null);
+            Class<?> c = Class.forName(cn.substring(1, cn.length() - 1).replace('/', '.'), false, null);
             return true;
         } catch (ClassNotFoundException e) {
             return false;
         }
     }
 
-    static final String      DONTINLINE_SIG = className("Ljdk/internal/vm/annotation/DontInline;");
-    static final String     FORCEINLINE_SIG = className("Ljdk/internal/vm/annotation/ForceInline;");
-    static final String          HIDDEN_SIG = className("Ljdk/internal/vm/annotation/Hidden;");
-    static final String INJECTEDPROFILE_SIG = className("Ljava/lang/invoke/InjectedProfile;");
-    static final String     LF_COMPILED_SIG = className("Ljava/lang/invoke/LambdaForm$Compiled;");
+    static final ClassDesc      DONTINLINE_SIG = classDesc("Ljdk/internal/vm/annotation/DontInline;");
+    static final ClassDesc     FORCEINLINE_SIG = classDesc("Ljdk/internal/vm/annotation/ForceInline;");
+    static final ClassDesc          HIDDEN_SIG = classDesc("Ljdk/internal/vm/annotation/Hidden;");
+    static final ClassDesc INJECTEDPROFILE_SIG = classDesc("Ljava/lang/invoke/InjectedProfile;");
+    static final ClassDesc     LF_COMPILED_SIG = classDesc("Ljava/lang/invoke/LambdaForm$Compiled;");
```

Comment:
These static fields can be made a `jdk.internal.classfile.Annotation`  instead of `ClassDesc` because that's what it ultimately represents.

```
   static final Annotation DONTINLINE      = annotation(DontInline.class);
     :
   private static Annotation annotation(Class<?> cls) {
        return Annotation.of(ClassDesc.ofDescriptor(cls.descriptorString()));
   }
```

Reply:
fixed, thanks.